src/vsr/replica.zig

```diff
@@ -879,9 +879,10 @@ pub fn ReplicaType(
             if (self.superblock.working.vsr_state.sync_op_max > 0) {
                 maybe(self.client_replies.writing.count() > 0);
                 for (0..constants.clients_max) |entry_slot| {
-                    if (self.client_sessions.entries_free.isSet(entry_slot)) {
-                        assert(!self.client_replies.faulty.isSet(entry_slot));
-                    } else {
+                    const slot_faulty = self.client_replies.faulty.isSet(entry_slot);
+                    const slot_free = self.client_sessions.entries_free.isSet(entry_slot);
+                    assert(!slot_faulty);
```

Comment:
Added this assert to catch the missing unset code. Before this commit, the code looks like it says that it actually `maybe(slot_faulty)` here, but it seems like that's not true?

We enter here after open or after state sync, and, in both cases, the faulty bits are reset. And the sessions are reset as well, so nothing should be able to _set_ this bit sneakily, even if we are reading some of the replies asynchronously?